### PR TITLE
Allow chef-solo to consume release tarball from github

### PR DESCRIPTION
**Pull request by ranjib**

### CODE REVIEW

**Comment by lamont-granquist** [comment]
Reiterating what I mentioned on the RFC:

We already decided to replace chef-solo with local-mode, adding more unique features to solo just makes the parity problem worse: https://github.com/chef/chef-rfc/blob/master/rfc031-replace-solo-with-local-mode.md

File: lib/chef/application/client.rb [hunk 1]
```diff
@@ -280,7 +287,14 @@ def reconfigure
       FileUtils.mkdir_p(Chef::Config.chef_repo_path)
       tarball_path = File.join(Chef::Config.chef_repo_path, 'recipes.tgz')
       fetch_recipe_tarball(Chef::Config[:recipe_url], tarball_path)
-      result = shell_out!("tar zxvf #{tarball_path} -C #{Chef::Config.chef_repo_path}")
+      if Chef::Config[:github]
+        cookbooks_dir = File.join(Chef::Config.chef_repo_path, 'cookbooks')
+        FileUtils.rm_rf(cookbooks_dir, :secure => true)
```
Comment (lamont-granquist) [inline lib/chef/application/client.rb]: After an errant spec test wiped out my chef github checkout (and could have easily wiped out $HOME or other nastiness) I'm incredibly :-1: on this rm_rf being here.

See also http://www.theregister.co.uk/2015/01/17/scary_code_of_the_week_steam_cleans_linux_pcs/

**Comment by lamont-granquist** [comment]
Needs some specs as well.

File: lib/chef/application/client.rb (hunk 1 above)
Comment (nathenharvey) [inline lib/chef/application/client.rb]: Can we do something other than `rm_rf`?!  that feels wrong and likely to cause someone to have a really bad day.

File: lib/chef/application/solo.rb [hunk 2]
```diff
@@ -207,7 +214,14 @@ def reconfigure
       FileUtils.mkdir_p(recipes_path)
       tarball_path = File.join(recipes_path, 'recipes.tgz')
       fetch_recipe_tarball(Chef::Config[:recipe_url], tarball_path)
-      Chef::Mixin::Command.run_command(:command => "tar zxvf #{tarball_path} -C #{recipes_path}")
+      if Chef::Config[:github]
+        cookbooks_dir = File.join(recipes_path, 'cookbooks')
+        FileUtils.rm_rf(cookbooks_dir, :secure => true)
```
Comment (nathenharvey) [inline lib/chef/application/solo.rb]: Can we do something other than `rm_rf`?!  that feels wrong and likely to cause someone to have a really bad day.

**Comment by ranjib** [comment]
i was reusing the existing chef-solo style, it has the FileUtils.rm_rf invocation[1]. I'll get rid of it.
@lamont-granquist you want a full blown functional test or just a unit test for the logic with untar and fetch being stubbed?
[1]https://github.com/chef/chef/blob/master/lib/chef/application/solo.rb#L205

**Comment by lamont-granquist** [comment]
I *like* both functional and unit tests.  I'll take whichever I can get =)

If the functional test looks like a yak shave (particularly with windows) then just the unit test would be fine.

**Comment by ranjib** [comment]
@lamont-granquist added spec

**Comment by ranjib** [comment]
closing this. discussion around this available on the RFC (now closed). I'll revisit this as part of a wider bootstrap overhaul RFC

**Pull request closed without merging**